### PR TITLE
Dragging - Improve `featureCamera` handling

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- Upon entering an arsenal, the player will drop his dragged or carried object.
- Applies to both BI and ACE arsenals.
- Only applies if the player opens the arsenal on himself.

The right click pop-up is annoying and distracting if you enter an arsenal.

That's why I'm thinking of something similar for when entering the Zeus interface, especially if you use ZEN which uses right clicks. Thoughts on that idea?

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by LinkIsGrim** [comment]
`featureCamera` CBA vent is more generic, if you'd like to use that.

**Comment by johnb432** [comment]
> `featureCamera` CBA event is more generic, if you'd like to use that.

Updated to include that change. Are there any cameras that should allow dragging/carrying?

**Comment by jonpas** [comment]
Splendid Camera should probably allow it.

**Comment by LinkIsGrim** [comment]
> > `featureCamera` CBA event is more generic, if you'd like to use that.
> 
> Updated to include that change. Are there any cameras that should allow dragging/carrying?

Player probably shouldn't drop items when opening Zeus, either.

**Comment by rautamiekka** [comment]
> > > `featureCamera` CBA event is more generic, if you'd like to use that.
> > 
> > 
> > Updated to include that change. Are there any cameras that should allow dragging/carrying?
> 
> Player probably shouldn't drop items when opening Zeus, either.

Yeah, definitely not, will cause more problems than solves/prevents.

**Comment by johnb432** [comment]
> Player probably shouldn't drop items when opening Zeus, either.

Any particular reason why?

Currently if you are dragging/carrying and go into Zeus, the mouse button hint is in the middle of the screen - not a huge deal, but unnecessary and annoying imo.
If you want to move your camera, you will drop the dragged/carried object anyway. I don't understand why you would want to limit yourself by not right-clicking, just to keep the dragged/carried object. Imo far easier to drop and pick up again afterwards.

---

An entirely different approach to this problem could be to "pause" dragging/carrying whilst in any camera other than default. By "pause" I mean remove the mouse button hint and remove the mouse button EH. When getting back into the default camera, those would be added back.

I'm heavily leaning towards this new approach, would be more elegant imo.

**Comment by rautamiekka** [comment]
Dropping could have 2 effects:
1. Tip off the observing players to be unnecessarily cautious, cuz at minimum it looks strange that the Zeus drops something, leaves it there for at least a bit, then picks it up again, and likely rinse-and-repeat, as he's using the interface to see what the Players are doing.
2. Cause a script depending on the Zeus carrying/dragging to break, malfunction, or fire early.

In other words it would hurt or break the story. I don't see any reasons for this for the Zeus interface.

----
Yeah, the "pause" is a better way.

**Comment by johnb432** [comment]
Pause has been added.

**Review by LinkIsGrim — APPROVED** [review]
